File: lib/src/utils/string_utils.dart
class StringUtils {
  static RegExp _emptyRegex = RegExp(r'^\s*$');
  static bool isNullOrEmpty(String? value,
      {bool considerWhiteSpaceAsEmpty = false}) {
    if (considerWhiteSpaceAsEmpty)
      return value == null || _emptyRegex.hasMatch(value);
    return value?.isEmpty ?? true;
  }
}

File: ios/Classes/lib/awesome_notifications_core/utils/StringUtils.swift
//
//  StringUtils.swift
//  awesome_notifications
//
//  Created by Rafael Setragni on 05/09/20.
//

import Foundation

public class StringUtils {

    public static func isNullOrEmpty(_ value:String?) -> Bool {
        return value?.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty ?? true
    }
    
    public static func random(length: Int) -> String {
        let letters : NSString = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
        let len = UInt32(letters.length)

        var randomString = ""

        for _ in 0 ..< length {
            let rand = arc4random_uniform(len)
            var nextChar = letters.character(at: Int(rand))
            randomString += NSString(characters: &nextChar, length: 1) as String
        }

        return randomString
    }
}
